src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+
+@Component({
+  selector: 'text-input',
+  templateUrl: 'text-input.component.html'
+})
+export class TextInputComponent implements OnInit {
+  @Output()
+  valueChange: EventEmitter<string> = new EventEmitter<string>()
+
+  @Input()
+  type
+
+  placeholder = 'Enter text'
+  isDateTime: string
+  dateFormat: string
+
+  ngOnInit() {
+    this.isDateTime = this.getIsDateTime()
+    if (this.isDateTime) this.dateFormat = this.getDateFormat()
+  }
+
+  getIsDateTime() {
```

Comment:
Please rename it to `isDateOrTime` otherwise one might confuse it with `datetime_*` in redcap.

Reply:
Made the changes.